code/game/objects/items/crab17.dm

```diff
@@ -14,7 +14,7 @@
 	if(activated)
 		to_chat(user, "<span class='warning'>You already activated Protocol CRAB-17.</span>")
 		return FALSE
-	if(alert(user, "Are you sure you want to crash this market with no survivors?", "Protocol CRAB-17", "Yes", "No") == "Yes")
+	if(tgui_alert(user, "Are you sure you want to crash this market with no survivors?", "Protocol CRAB-17", list("Yes", "No")) == "Yes")
```

Comment:
Any alerts using `== (True condition for dangerous operation)` should change to `!= (false condition for dangerous operation)`

If hitting Yes does nothing, then it should be that as the true condition. This is so if the window is closed (causing a null return) it will not execute. It should always do the minimum action when the window is closed.